core/src/Kokkos_Threads.hpp

```diff
@@ -107,8 +108,13 @@ class Threads {
   /// method does not return until all dispatched functors on this
   /// device have completed.
   static void impl_static_fence();
+  static void impl_static_fence(const std::string& name);
+
+  void fence(Impl::fence_is_static is_static) const;
```

Comment:
what are these new overloads, and why are there public functions which only exist for this backend?

Reply:
These functions exist for any backend on which there was no separation between a "static" fence and an "instance" fence. In the past, an instance fence was the same thing as a static fence, in terms of implementation. The ugly part of this PR is that while these are still mechanically the same thing, we need to be able to differentiate them for profiling. So I basically just wrote these to provide the plumbing I needed to tell the tools subsystem about a "Whole Device" fence or "instance fence." These still mechanically do the same thing, except in their reporting.

I could probably make the instance fence that takes in `is_static` into an impl method, but I was kinda leaning on "functions with Impl args are Impl." But if we want to make the function itself impl, I'm not too hostile to that

Reply:
I would feel better with that then being an internal function. We already distinguish between the static fence and the non-static fence technically: impl_static_fence and fence, with the static version only being there for internal purposes such as for the implementation of Kokkos::fence(); So if the reason right now is that impl_static_fence() calls fence(), then just make them both call somethign internal.